File: practices/exercism/c/2020/raindrops.c
/* Raindrops from https://exercism.io */

#include "raindrops.h"
#include <stdio.h>
#include <string.h>

char *convert(char result[], int drops)
{
  if (drops % 3 == 0) {
    strcat(result, "Pling");
  }

  if (drops % 5 == 0) {
    strcat(result, "Plang");
  }

  if (drops % 7 == 0) {
    strcat(result, "Plong");
  }

  if (drops % 3 != 0 && drops % 5 != 0 && drops % 7 != 0) {
    sprintf(result, "%d", drops);
  }

  return result;
}

File: sandbox/exercism/ruby/2019/raindrops.rb
# Raindrops from https://exercism.io

class Raindrops
  SOUNDS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }.freeze

  def self.convert(number)
    new(number).convert
  end

  def initialize(number)
    @number = number
  end

  def convert
    SOUNDS.inject('') { |result, (k, v)| (number % k).zero? ? result << v : result }
          .then { |sounds| sounds.empty? ? number.to_s : sounds }
  end

  private

    attr_reader :number
end
